pkg/cloudprovider/providers/photon/photon.go

```diff
@@ -0,0 +1,550 @@
+	projID, err := getProjIDbyName(cfg.Global.Tenant, cfg.Global.Project)
+	if err != nil {
+		return nil, logError("getProjIDbyName failed when creating new Photon Controller client", err)
+	}
+
+	// Get local hostname for localInstanceID
+	cmd := exec.Command("bash", "-c", `echo $HOSTNAME`)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		return nil, logError("get local hostname bash command failed", err)
+	}
+	if out.Len() == 0 {
+		return nil, logError("unable to retrieve hostname for Instance ID", nil)
+	}
+	hostname := strings.TrimRight(out.String(), "\n")
+	vmID, err := getVMIDbyNodename(projID, hostname)
+	if err != nil {
+		return nil, logError("getVMIDbyNodename failed when creating new Photon Controller client", err)
+	}
+
+	pc := PCCloud{
+		cfg:              &cfg,
+		localInstanceID:  vmID,
+		localHostname:    hostname,
+		localK8sHostname: "",
+		projID:           projID,
+	}
+
+	overrideIP = cfg.Global.OverrideIP
+
+	return &pc, nil
+}
+
+// Instances returns an implementation of Instances for Photon Controller.
+func (pc *PCCloud) Instances() (cloudprovider.Instances, bool) {
+	return pc, true
+}
+
+// List is an implementation of Instances.List.
+func (pc *PCCloud) List(filter string) ([]k8stypes.NodeName, error) {
+	return nil, nil
+}
+
+// NodeAddresses is an implementation of Instances.NodeAddresses.
+func (pc *PCCloud) NodeAddresses(nodeName k8stypes.NodeName) ([]api.NodeAddress, error) {
+	addrs := []api.NodeAddress{}
+	name := string(nodeName)
+
+	var vmID string
+	var err error
+	if name == pc.localK8sHostname {
+		vmID = pc.localInstanceID
+	} else {
+		vmID, err = getInstanceID(name, pc.projID)
+		if err != nil {
+			return addrs, logError("getInstanceID failed for NodeAddresses", err)
+		}
+	}
+
+	// Retrieve the Photon VM's IP addresses from the Photon Controller endpoint based on the VM ID
+	vmList, err := Esxclient.Projects.GetVMs(pc.projID, nil)
+	if err != nil {
+		return addrs, fmt.Errorf("Photon Cloud Provider: Failed to GetVMs for project %s, error [%v]", pc.projID, err)
+	}
+
+	for _, vm := range vmList.Items {
+		if vm.ID == vmID {
+			task, err := Esxclient.VMs.GetNetworks(vm.ID)
+			if err != nil {
+				return addrs, logError("GetNetworks failed for node "+name+" with vm.ID "+vm.ID, err)
+			} else {
+				task, err = Esxclient.Tasks.Wait(task.ID)
+				if err != nil {
+					return addrs, logError("Wait task for GetNetworks failed for node"+name+" with vm.ID "+vm.ID, err)
+				} else {
+					networkConnections := task.ResourceProperties.(map[string]interface{})
+					networks := networkConnections["networkConnections"].([]interface{})
+					for _, nt := range networks {
+						network := nt.(map[string]interface{})
+						if val, ok := network["ipAddress"]; ok && val != nil {
+							ipAddr := val.(string)
+							if ipAddr != "-" {
+								api.AddToNodeAddresses(&addrs,
+									api.NodeAddress{
+										// TODO: figure out the type of the IP
```

Comment:
Do you need to do this now? 

I figure out the type of IP address by looking at the MAC address. Let me know if you need details.